src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java

```diff
@@ -1081,13 +1081,34 @@ static String getRanges(JsonArray ar, String startProp, String endProp) {
         });
   }
 
-  /** Draft, Work in Progress (WIP). */
   Future<Void> getUseOverTime(Vertx vertx, RoutingContext ctx) {
+    String format = ctx.request().params().get("format");
+    switch (format) {
+      case "JOURNAL": return getUseOverTimeJournal(vertx, ctx);
+      case "BOOK": return getUseOverTimeBook(vertx, ctx);
+      case "DATABASE": return getUseOverTimeDatabase(vertx, ctx);
+      default: throw new IllegalArgumentException("format = " + format);
+    }
+  }
+
+  Future<Void> getUseOverTimeJournal(Vertx vertx, RoutingContext ctx) {
     TenantPgPool pool = TenantPgPool.pool(vertx, TenantUtil.tenant(ctx));
     String agreementId = ctx.request().params().get("agreementId");
     String start = ctx.request().params().get("startDate");
     String end = ctx.request().params().get("endDate");
 
+    return getUseOverTimeJournal(pool, agreementId, start, end)
+        .map(json -> {
+          ctx.response().setStatusCode(200);
+          ctx.response().putHeader("Content-Type", "application/json");
+          ctx.response().end(json.encodePrettily());
+          return null;
+        });
+  }
+
+  Future<JsonObject> getUseOverTimeJournal(TenantPgPool pool,
```

Comment:
It's hard for me to think that it's a good idea to give these two functions the same name. Overloading makes sense when the same-named functions are doing the same thing, but that's not the case here.

Reply:
They do the same thing.
This one takes the parameters out of ctx, calls the other one, and then returns the result using ctx.